src/v/datalake/tests/datalake_protobuf_tests.cc

```diff
@@ -0,0 +1,410 @@
+    topic_1.mutable_partitions()->Add(
+      make_partition(1, {{12, 1}, {11, 0}, {13, 4}}));
+    topic_1.mutable_partitions()->Add(
+      make_partition(3, {{11, 3}, {12, 1}, {13, 2}}));
+
+    message.mutable_topics()->emplace("topic_1", std::move(topic_1));
+
+    Topic topic_2;
+    topic_2.set_name("topic_2");
+    topic_2.mutable_partitions()->Add(make_partition(0, {{11, 0}}));
+    message.mutable_topics()->emplace("topic_2", std::move(topic_2));
+
+    auto result = co_await serialize_and_convert(message);
+
+    ASSERT_TRUE_CORO(result.has_value());
+    auto optional_value = std::move(result.value());
+    ASSERT_TRUE_CORO(result.value().has_value());
+    auto result_value = std::get<std::unique_ptr<struct_value>>(
+      std::move(optional_value.value()));
+
+    EXPECT_THAT(
+      result_value->fields,
+      ElementsAre(
+        IcebergMap(UnorderedElementsAre(
+          IcebergKeyValue(
+            IcebergPrimitive<string_value>("topic_2"),
+            // topic
+            IcebergStruct(
+              OptionalIcebergPrimitive<string_value>("topic_2"),
+              IcebergList(ElementsAre(
+                // partition
+                IcebergStruct(
+                  OptionalIcebergPrimitive<int_value>(0),
+                  IcebergList(ElementsAre(
+                    // broker shard
+                    IcebergStruct(
+                      OptionalIcebergPrimitive<iceberg::long_value>(11),
+                      OptionalIcebergPrimitive<iceberg::int_value>(0)))),
+                  OptionalIcebergPrimitive<int_value>(0)))))),
+          IcebergKeyValue(
+            IcebergPrimitive<string_value>("topic_1"),
+            // topic
+            IcebergStruct(
+              OptionalIcebergPrimitive<string_value>("topic_1"),
+              IcebergList(ElementsAre(
+                // partition
+                IcebergStruct(OptionalIcebergPrimitive<int_value>(0), _, _),
+                // partition
+                IcebergStruct(OptionalIcebergPrimitive<int_value>(1), _, _),
+                // partition
+                IcebergStruct(
+                  OptionalIcebergPrimitive<int_value>(3), _, _))))))),
+        IcebergList(ElementsAre(
+          OptionalIcebergPrimitive<long_value>(11),
+          OptionalIcebergPrimitive<long_value>(12),
+          OptionalIcebergPrimitive<long_value>(13)))));
+}
+
+TEST_CORO(values_protobuf, TestEmptyMessage) {
+    protobuf_test_messages::editions::TestAllTypesEdition2023 test_msg;
+    // Make sure that empty message is parsable
+    auto result = co_await serialize_and_convert(test_msg);
+    ASSERT_TRUE_CORO(result.has_value());
+
+    auto iceberg_value = std::move(result.value().value());
+
+    auto st_value = std::get<std::unique_ptr<iceberg::struct_value>>(
+      std::move(iceberg_value));
+
+    for (auto& f : st_value->fields) {
+        // in an empty message only primitives should have values, the other
+        // fields must be set to empty optional
```

Comment:
Can we assert this? I don't see any tests asserting empty messages are converted into `nullopt`.

Reply:
I think this test implicitly verifies that, here we assert that the present field is an `iceberg_primitve` since the test message has a lot of other fields not being primitives they are empty optionals otherwise the assert:
```c++
if (f.has_value()) {
            ASSERT_TRUE_CORO(
              std::holds_alternative<iceberg::primitive_value>(*f));
}
```
would fail

Reply:
i've added more explicit tests for the presence checking